src/game/room.c

```diff
@@ -977,3 +981,66 @@ void Room_TestTriggers(int16_t *data, bool heavy)
         }
     }
 }
+
+VECTOR *Room_GetWalkableHeights(
+    FLOOR_INFO *floor, int32_t x, int32_t y, int32_t z)
+{
+    ROOM_INFO *r;
+    while (floor->pit_room != NO_ROOM) {
+        ROOM_INFO *r = &g_RoomInfo[floor->pit_room];
+        int32_t x_floor = (z - r->z) >> WALL_SHIFT;
+        int32_t z_floor = (x - r->x) >> WALL_SHIFT;
+        floor = &r->floor[x_floor + z_floor * r->x_size];
+    }
+
+    int16_t *floor_data = &g_FloorData[floor->index];
+    int16_t type;
+    int16_t trigger;
+    int16_t trig_flags;
+    int16_t trig_type;
+    VECTOR *heights = Vector_Create(sizeof(int32_t));
+
+    do {
+        type = *floor_data++;
+
+        switch (type & DATA_TYPE) {
+        case FT_TILT:
+        case FT_ROOF:
+        case FT_DOOR:
+            floor_data++;
+            break;
+
+        case FT_LAVA:
+            break;
+
+        case FT_TRIGGER:
+            trig_flags = *floor_data;
+            floor_data++;
+            trig_type = (type >> 8) & 0x3F;
+            do {
+                trigger = *floor_data++;
+                if (TRIG_BITS(trigger) != TO_OBJECT) {
+                    if (TRIG_BITS(trigger) == TO_CAMERA) {
+                        trigger = *floor_data++;
+                    }
+                } else {
```

Comment:
Ok. I think I reused code from the stats function here which does it like this.

I had a wish to have a common floor traversal function since they are all slightly different and repeated, but each time it happens there are different things being done while traversing. Stats sets killables, secrets, and pickups, injection does its own thing, etc.